---
title: "Gene Coexpression Networks"
excerpt: "<br/><img src='/images/networks.png'>"
collection: portfolio
---

Coexpression networks help overcome the limitations of sparsity in gene functional annotations, by providing a means to transfer knowledge from annotated genes to genes yet unannotated, based on the 'guilt-by association' assumptions. 
[RECoN](https://plantstress-pereira.uark.edu/RECoN/){: .btn .btn-outline} and [SANe](https://plantstress-pereira.uark.edu/SANe/){: .btn .btn-outline} are designed to identify clusters of functionally related genes that tightly coexpress in a compendium of abiotic-stress gene expression experiments in rice and seed development stages in Arabidopsis, respectively. Both these tools allow users to upload and analyze results from relevant RNA-seq experiments in context of the underlying coexpression networks. The tools output list of clusters that changed most in the uploaded RNA-seq data. Novel genes in the top clusters become a set of candidates for further exploration. 


## Publications:
1. Recent advances in gene function prediction using context-specific coexpression networks in plants. Chirag Gupta and Andy Pereira,  F1000 Research, 2019. [url](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6364378/)
2. SANe: The Seed Active Network for Discovering Transcriptional Regulatory Programs of Seed Development. Gupta et al., biorxiv, 2018. [url](https://www.biorxiv.org/content/10.1101/165894v2)
3. RECoN: rice environment coexpression network for systems level analysis of abiotic-stress response. A Krishnan, C Gupta, MMR Ambavaram, A Pereira. Frontiers in Plant Science, 2017. [url](https://www.frontiersin.org/articles/10.3389/fpls.2017.01640/full)
4. Transcriptome-based Gene Networks for Systems-level Analysis of Plant Gene Functions. Chirag Gupta.  University of Arkansas, 2017. [url](https://scholarworks.uark.edu/etd/2526/)
